### PR TITLE
Add a "cluster curl" tool

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
This adds a curl-like command to our cli, called `clusters rest-call`. The user specifies which cluster and then a path, and it makes a call and returns the result.

```
(.venv) sh-5.2# console clusters curl target_cluster _cat/indices
green  open .opensearch-observability    ZZV9G4p_QbW9Z4IMzis0cw 1 0  0 0   208b   208b
green  open .plugins-ml-config           4-zA4PGQSG2Phx_hm8H1Ng 1 0  1 0  3.9kb  3.9kb
yellow open security-auditlog-2024.10.04 yZhCHEXUScOmmn4BipEWeg 1 1  3 0   50kb   50kb
yellow open new-index                    xunLH4XMRXqhiKS0viY0IQ 1 1  3 0  8.4kb  8.4kb
green  open .opendistro_security         gxQ9wlmkQHCvJKodmavnRA 1 0 10 0 78.3kb 78.3kb

(.venv) sh-5.2# 
(.venv) sh-5.2# 
(.venv) sh-5.2# 
(.venv) sh-5.2# console clusters curl target_cluster /
{
  "name" : "13d4b5b49ff8",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "B-TtUe80Svu5NiPISo_47A",
  "version" : {
    "distribution" : "opensearch",
    "number" : "2.15.0",
    "build_type" : "tar",
    "build_hash" : "61dbcd0795c9bfe9b81e5762175414bc38bbcadf",
    "build_date" : "2024-06-20T03:27:32.562036890Z",
    "build_snapshot" : false,
    "lucene_version" : "9.10.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}

(.venv) sh-5.2# 
(.venv) sh-5.2# 
(.venv) sh-5.2# 
(.venv) sh-5.2# console clusters curl -XPOST target_cluster /new-index-2/_doc --json '{"id": 3, "number": 10}'
{"_index":"new-index-2","_id":"tCzFbJIBO46AuR2VLwFV","_version":1,"result":"created","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":0,"_primary_term":1}
(.venv) sh-5.2#
```

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-2047

### Testing
Adding now.

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/Pipfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[[source]]
```
Comment (peternied) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/Pipfile:1]: Why not use reuse the same workflow we have in clusters tool which uses `cluster.call_api(...)`?

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/cli.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            raise click.BadParameter(f"Invalid header format: {h}. Expected format: 'Header: Value'.")
+    return headers
+
+
+@click.command()
+@click.option(
+    "--config-file", default="/etc/migration_services.yaml", help="Path to config file"
+)
+@click.option('-v', '--verbose', count=True, help="Verbosity level. Default is warn, -v is info, -vv is debug.")
+@click.option('-X', '--request', default='GET', help="HTTP method to use",
```
Comment (gregschohn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/cli.py:48]: --request is a bit confusing.  Is that what curl (or anything else) uses?  'Verb' or 'method' are common labels

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/setup.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+setup(
+    name="cluster_curl",
+    version="1.0.0",
+    description="A Python tool to make API calls against congfigured clusters",
+    packages=find_packages(exclude=("tests")),
+    install_requires=["console_link", "Click"],
+    entry_points={
+        "console_scripts": [
+            "ccurl = cli:cli",
```
Comment (gregschohn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/setup.py:11]: I'd choose another name other than curl as the root name.  With curl, a user might think that they can use --verbose to trace activity, append a binary file,  set cookies, etc.  This is useful by itself.  It doesn't need to try to replace or compete with curl.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/cli.py (hunk 2 above)
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/cli.py:48]: This is what curl uses

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/Pipfile (hunk 1 above)
Comment (peternied) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/cluster_curl/Pipfile:1]: Oh, I missed the usage case at the top or your description, my apologyizes.

I was expecting our users to be able to run `console cluster curl target -XGET /foo/bar` - any reason why not? Seems like starting with `console ...` is a way to organize and group the MA functionality.

**Review by peternied — APPROVED** [review]

**Pull request merged**